tests/testsuite/publish.rs

```diff
@@ -4509,3 +4509,38 @@ Caused by:
 "#]])
         .run();
 }
+
+#[cargo_test]
+fn publish_reads_token_from_stdin() {
+    let registry = RegistryBuilder::new()
+        .alternative_named("i-need-token")
+        .http_api()
+        .token(registry::Token::Plaintext("TOKEN".to_string()))
+        .auth_required()
+        .build();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                edition = "2024"
+            "#,
+        )
+        .file("src/main.rs", "fn main() {}")
```

Comment:
nit: use lib target to make test execution time a bit better.

```suggestion
        .file("src/lib.rs", "")
```

Reply:
Thank you for the tip 😺